mobgap/sl/_stride_length.py

```diff
@@ -0,0 +1,296 @@
+        duration = int(np.floor(len(vacc) / sampling_rate_hz))  # bottom-rounded duration (s)
+        ICtime = initial_contacts / sampling_rate_hz  # initial contacts: samples -> sec
+        sec_centers = (
+            np.arange(0, duration) + 1
+        )  # TODO: Differently from how it is done for cadence, here we consider second bins going from 0.5 to 1.5, from 1.5 to 2.5, etc...
+        # padding last value of the calculated step length values so that sl_zjilstra_v3 is as long as ICtime
+        if len(sl_zjilstra_v3) < len(ICtime):
+            sl_zjilstra_v3 = np.concatenate(
+                [sl_zjilstra_v3, np.tile(sl_zjilstra_v3[-1], len(ICtime) - len(sl_zjilstra_v3))]
+            )
+
+        slSec_zjilstra_v3 = robust_step_para_to_sec(
+            ICtime, sl_zjilstra_v3, sec_centers, self.max_interpolation_gap_s, self.step_length_smoothing.clone()
+        )
+        # 11. Approximated stride length per second values = 2* step length per second values
+        slSec = slSec_zjilstra_v3[0:duration]*2
+
+        # Results
+        self.sl_sec_list_ = pd.DataFrame(
+            {"stride_length_m": slSec}, index=as_samples(sec_centers, sampling_rate_hz)
+        ).rename_axis(index="sec_center_samples")
+        self.stpl_list_ = sl_zjilstra_v3
+
+        return self
+
+
+def _zjilsV3(
+    vacc_high: np.ndarray,
+    sampling_rate_hz: float,
+    tuning_coefficient: float,
+    initial_contacts: pd.Series,
+    sensor_height: float,
+) -> np.ndarray:
+    """
+    Step length estimation using the biomechanical model propose by Zijlstra & Hof [1]
+    [1] Zijlstra, W., & Hof, A. L. (2003). Assessment of spatio-temporal gait parameters from trunk accelerations
+    during human walking. Gait & posture, 18(2), 1-10.
+    Inputs:
+    - vacc_high: vertical acceleration recorded on lower back, high-pass filtered.
+    - sampling_rate_hz: sampling frequency of the acc signal.
+    - tuning_coefficient: correction factor estimated by data from various clinical populations
+    - initial_contacts: vector containing the timing of initial contacts events (samples)
+    - sensor_height: Low Back height, i.e., the distance from ground to sensor location on lower back (in m)
+
+    Output:
+    - sl_zjilstra_v3: estimated step length.
+    """
+    # estimate vertical speed
+    vspeed = -np.cumsum(vacc_high) / sampling_rate_hz
+    # drift removal (high pass filtering)
+    fc = 1
+    b, a = butter(4, Wn=fc / (sampling_rate_hz / 2), btype="high", output="ba")  # TODO: why this causes a warning?
+    speed_high = lfilter(b, a, vspeed)
+    # estimate vertical displacement
+    vdis_high_v2 = np.cumsum(speed_high) / sampling_rate_hz
+    # initialize the output array as an array of zeros having one less element
+    # than the array of ICs
+    h_jilstra_v3 = np.zeros(len(initial_contacts) - 1)
+    # initial estimates of the step length
+    for k in range(len(initial_contacts) - 1):
```

Comment:
I think there is a "find_min_max" function implemented for one of the GSD or SL algorithms. We might be able to use this here.

Reply:
Ok thanks for the suggestion!

Reply:
I am sorry, I did not find anything like a find_min_max function... The most similar that I found is a function to find minima between zero crossings, which is called by the detect method of the IcdIonescu class